code/core/write_sdl_ntuple.cc

```diff
@@ -782,6 +843,195 @@ void setGnnNtupleMiniDoublet(SDL::Event* event, unsigned int MD)
+        {
+            delta1 = 1.f;//0.009;
+            delta2 = 1.f;//0.009;
+            module_slope = -999.f;
+            is_flat = true;
+        }
+        // Category 3: barrel PS tilted
+        else if (module_subdet == SDL::Barrel and module_type == SDL::PS and module_side != SDL::Center)
+        {
+
+            delta1 = inv1;//1.1111f;//0.01;
+            is_flat = false;
+
+            delta2 = (inv2 * module_drdz/sqrtf(1 + module_drdz * module_drdz));
+        }
+        // Category 4: endcap PS
+        else if (module_subdet == SDL::Endcap and module_type == SDL::PS)
+        {
+            delta1 = inv1;//1.1111f;//0.01;
+            is_flat = false;
+
+            /* Despite the type of the module layer of the lower module index,
+             * all anchor hits are on the pixel side and all non-anchor hits are
+             * on the strip side! */
+            delta2 = inv2;//16.6666f;//0.15f;
+        }
+        // Category 5: endcap 2S
+        else if (module_subdet == SDL::Endcap and module_type == SDL::TwoS)
+        {
+            delta1 = 1.f;//0.009;
+            delta2 = 500.f*inv1;//555.5555f;//5.f;
+            is_flat = false;
+        }
+        else
+        {
+            printf("ERROR!!!!! I SHOULDN'T BE HERE!!!! subdet = %d, type = %d, side = %d\n", module_subdet, module_type, module_side);
+        }
+
+        /* Since C++ can't represent infinity, SDL_INF = 123456789 was used 
+         * to represent infinity in the data table */
+        float absArctanSlope = ((module_slope != 123456789) ? fabs(atanf(module_slope)) : 0.5f*float(M_PI));
+
+        float hit_x = hitsInGPU.xs[hit];
+        float hit_y = hitsInGPU.ys[hit];
+        float angleM;
+        if (hit_x > 0 and hit_y > 0)
+        {
+            angleM = 0.5f*float(M_PI) - absArctanSlope;
+        }
+        else if (hit_x < 0 and hit_y > 0)
+        {
+            angleM = absArctanSlope + 0.5f*float(M_PI);
+        }
+        else if (hit_x < 0 and hit_y < 0)
+        {
+            angleM = -(absArctanSlope + 0.5f*float(M_PI));
+        }
+        else if (hit_x > 0 and hit_y < 0)
+        {
+            angleM = -(0.5f*float(M_PI) - absArctanSlope);
+        }
+
+        float xPrime, yPrime;
+        if(not is_flat)
+        {
+            xPrime = hit_x * cosf(angleM) + hit_y * sinf(angleM);
+            yPrime = hit_y * cosf(angleM) - hit_x * sinf(angleM);
+        }
+        else
+        {
+            xPrime = hit_x;
+            yPrime = hit_y;
+        }
+        sigmas.push_back(2 * sqrtf((xPrime * delta1) * (xPrime * delta1) + (yPrime * delta2) * (yPrime * delta2)));
+    }
+
+    ana.tx->pushbackToBranch<float>("t3_0_dr", sigmas[0]);
+    ana.tx->pushbackToBranch<float>("t3_2_dr", sigmas[1]);
+    ana.tx->pushbackToBranch<float>("t3_4_dr", sigmas[2]);
+
+    // Constants
+    const float kRinv1GeVf = (2.99792458e-3 * 3.8);
+    const float k2Rinv1GeVf = kRinv1GeVf / 2.;
```

Comment:
Should we just include the constants from [here](https://github.com/SegmentLinking/TrackLooper/blob/579b9d08bfb5d774c6ce3727e7625a4620faf1ac/SDL/Constants.h#L78)?